augur/filter/__init__.py

```diff
@@ -36,9 +36,13 @@ def register_arguments(parser):
     metadata_filter_group.add_argument('--exclude-where', nargs='+',
                                 help="Exclude samples matching these conditions. Ex: \"host=rat\" or \"host!=rat\". Multiple values are processed as OR (matching any of those specified will be excluded), not AND")
     metadata_filter_group.add_argument('--exclude-all', action="store_true", help="exclude all strains by default. Use this with the include arguments to select a specific subset of strains.")
-    metadata_filter_group.add_argument('--include', type=str, nargs="+", help="file(s) with list of strains to include regardless of priorities or subsampling")
-    metadata_filter_group.add_argument('--include-where', nargs='+',
-                                help="Include samples with these values. ex: host=rat. Multiple values are processed as OR (having any of those specified will be included), not AND. This rule is applied last and ensures any sequences matching these rules will be included.")
+    metadata_filter_group.add_argument('--include', type=str, nargs="+", help="file(s) with list of strains to include regardless of priorities, subsampling, or absence of an entry in --sequences.")
+    metadata_filter_group.add_argument('--include-where', nargs='+', help="""
+        Include samples with these values. ex: host=rat. Multiple values are
+        processed as OR (having any of those specified will be included), not
+        AND. This rule is applied last and ensures any strains matching these
+        rules will be included regardless of priorities, subsampling, or absence
```

Comment:
> This rule is applied last and ensures any strains matching these rules will be included regardless of priorities, subsampling, or absence of an entry in --sequences

Is there any way to use `--include-where` _and_ restrict output metadata to those with sequences?

More generally, if we provide `--sequences`, what's the use case to ever export metadata which doesn't have a corresponding sequence?

Reply:
Good point. I briefly thought about that but decided to be conservative in this PR, aiming to clarifying the existing behavior.

Currently, the code applies exclusions followed by inclusions. Moving the missing sequence exclusionary filter to be applied after inclusions would be a bigger change that can be considered as another issue/PR.

Reply:
Thanks for adding the updated arg description in b705f6f9de7cfcc4877c1c4751c103897036ac02 👍